src/modules/devicemodewatcher/devicemodewatcher.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-FileCopyrightText: 2022-2022 liulinsong <liulinsong@kyllinos.cn>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ *
+ */
+
+#include "fcitx/addonfactory.h"
+#include "devicemodewatcher.h"
+
+namespace fcitx {
+
+DeviceModeWatcher::DeviceModeWatcher(Instance *instance)
+    : instance_(instance), bus_(dbus()->call<IDBusModule::bus>()),
+      watcher_(*bus_) {
+    entry_ = watcher_.watchService(
+        "com.kylin.statusmanager.interface",
```

Comment:
Please put all the new addon in to separate PR.

Reply:
I will seperate the new addon into new PR.